### PR TITLE
CI: adding valgrind job to Nightly

### PR DESCRIPTION
**Pull request by samir-nasibli**
### Description 
This PR was created for initial testing with valgrind. The developments from here will be analyzed and presented for discussion for further implementation.

- [ ] I have reviewed my changes thoroughly before submitting this pull request.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have updated the documentation to reflect the changes, if necessary (updated in # - _add PR number_)
- [ ] The unit tests pass successfully.
- [ ] I have run it locally and tested the changes extensively.
- [ ] I have resolved any merge conflicts that might occur with the base branch.
- [ ] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/intel/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_
- [ ] I have added a respective label(s) to PR if I have a permission for that.

### CODE REVIEW

File: .ci/pipeline/nightly.yml [hunk 1]
```diff
@@ -92,6 +92,46 @@ jobs:
...
+      ./conda-recipe/build.sh
+      python setup_sklearnex.py install --single-version-externally-managed --record=record.txt
+    displayName: 'Build daal4py/sklearnex'
+  - script: |
+      . /usr/share/miniconda/etc/profile.d/conda.sh
+      conda activate CB
+      cd ..
+      export PYTHONMALLOC=malloc
+      valgrind --show-leak-kinds=definite --log-file=/tmp/valgrind-output  \
+               python -m pytest --verbose --pyargs sklearnex               \
```
Comment (samir-nasibli) [inline .ci/pipeline/nightly.yml:130]: Make sense check only for specific test module or case

**Comment by ethanglaser** [comment]
Thanks, I've added this PR link to my ticket. Will re-open when I resume work on it.

**Pull request closed without merging**